### ISSUE TITLE
Add a warning or confirmation prompt when conecting to a new remote

### ISSUE THREAD

**Issue by vxid** [magit/magit#3969]
Sorry in advance if this was answered elsewhere but I could not find a similar request in other issues.

Magit hangs when pushing to a new remote because ssh is asking confirmation to connect to the remote host.

It would be nice to be able to answer the confirmation from within Magit or at least issue a warning in the magit-status screen.

**Comment by tarsius** [comment]
Magit usually notices when some subprocesses requests that the user provides a password. In order to do so it checks their output for known password prompts. If this does not work, then the problem likely is that the password prompt in question is not recognized.

So push on the command line and take note of the prompt. Then look at `magit-process-password-prompt-regexps` and figure out what has to be changed/added, and let us know about it.

**Comment by vxid** [comment]
The issue is not due to a password prompt, but to a host that is not yet present in `.ssh/known_hosts`. The exact prompt is 

```
The authenticity of host 'xxxxxx (xx.xxx.xxx.xx)' can't be established.
ECDSA key fingerprint is SHA256:xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx.
Are you sure you want to continue connecting (yes/no/[fingerprint])?
```

**Comment by tarsius** [comment]
Magit also tries to detect "yes or no" questions and forward them to the user.

`magit-process-yes-or-no-prompt-regexp` doesn't match this particular question because of the `/[fingerprint]`. Does it actually say `/[fingerprint]` or is it the same as the `xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx` above? Something else? I don't quite understand what this third choice stands for, what you have to enter to select it, and what happens if you do. How does it differ from "Yes"? Do you know more?

**Comment by vxid** [comment]
So I did some research. It seems this this new option was added recently to OpenSSH (`ssh -V` returns `OpenSSH_8.0p1, OpenSSL 1.1.1d  10 Sep 2019`). 

`fingerprint` is an option (written as-is) that only seems to work if the exact key fingerprint is used as an input. I still struggle to see what it can be used for but I am fairly ignorant of how OpenSSH works. The relevant function can be found around line 900 of `sshconnect.c` in the OpenSSH source. 

Here is the exact message below (xxx added to hide IP and hostname, the rests is as-is):

```
The authenticity of host 'gitlab.xxx.com (XX.XXX.XXX.XX)' can't be established.
ECDSA key fingerprint is SHA256:xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx.
Are you sure you want to continue connecting (yes/no/[fingerprint])? 
```
